hybridse/include/vm/physical_op.h

```diff
@@ -1470,6 +1491,64 @@ class PhysicalRequestUnionNode : public PhysicalBinaryNode {
     RequestWindowUnionList window_unions_;
 };
 
+class PhysicalRequestAggUnionNode : public PhysicalOpNode {
+ public:
+    PhysicalRequestAggUnionNode(PhysicalOpNode *request, PhysicalOpNode *raw, PhysicalOpNode *aggr,
+                             const RequestWindowOp &window,
+                             const RequestWindowOp &aggr_window,
+                             bool instance_not_in_window,
+                             bool exclude_current_time, bool output_request_row)
+        : PhysicalOpNode(kPhysicalOpRequestAggUnion, true),
+          window_(window),
+          agg_window_(aggr_window),
+          instance_not_in_window_(instance_not_in_window),
+          exclude_current_time_(exclude_current_time),
+          output_request_row_(output_request_row) {
+        output_type_ = kSchemaTypeTable;
+
+        fn_infos_.push_back(&window_.partition_.fn_info());
+        fn_infos_.push_back(&window_.sort_.fn_info());
+        fn_infos_.push_back(&window_.range_.fn_info());
+        fn_infos_.push_back(&window_.index_key_.fn_info());
+
+        AddProducers(request, raw, aggr);
+    }
+    virtual ~PhysicalRequestAggUnionNode() {}
+    base::Status InitSchema(PhysicalPlanContext *) override;
+    void Print(std::ostream &output, const std::string &tab) const override;
+    void PrintChildren(std::ostream& output, const std::string& tab) const override;
+    const bool Valid() { return true; }
+    static PhysicalRequestAggUnionNode *CastFrom(PhysicalOpNode *node);
+
+    const bool instance_not_in_window() const {
+        return instance_not_in_window_;
+    }
+    const bool exclude_current_time() const { return exclude_current_time_; }
+    const bool output_request_row() const { return output_request_row_; }
+    const RequestWindowOp &window() const { return window_; }
+
+    base::Status WithNewChildren(node::NodeManager *nm,
+                                 const std::vector<PhysicalOpNode *> &children,
+                                 PhysicalOpNode **out) override {
+        return base::Status(common::kUnSupport);
+    }
+
+    RequestWindowOp window_;
+    RequestWindowOp agg_window_;
+    const bool instance_not_in_window_;
```

Comment:
Since there are public member function, the public members is also need?

Reply:
actually no need. copy and paste from `PhysicalRequestUnionNode` :(